src/datasets/io/csv.py

```diff
@@ -0,0 +1,53 @@
+from typing import Optional
+
+from .. import Features, NamedSplit
+from ..packaged_modules.csv.csv import Csv
+from ..utils.typing import PathLike
+from .abc import AbstractDatasetReader
+
+
+class CsvDatasetReader(AbstractDatasetReader):
+    def __init__(
+        self,
+        path: PathLike,
+        split: Optional[NamedSplit] = None,
+        features: Optional[Features] = None,
+        cache_dir: str = None,
+        keep_in_memory: bool = False,
+        **kwargs,
+    ):
+        super().__init__(
+            path, split=split, features=features, cache_dir=cache_dir, keep_in_memory=keep_in_memory, **kwargs
+        )
+        self.builder = Csv(
+            cache_dir=cache_dir,
+            data_files=path,
```

Comment:
When data_files is just one path, then the builder generates one dataset that is considered to be the train split.
In order to use the `split` parameter of the Reader, you can change this by using a dict, like this:
```python
data_files={split: path}
```
The split defined in `data_files` are the ones you can get when using `as_dataset`

You may have to set a default value for `split` though (the builder does require a valid split name, not None):
```python
split = split if split is not None else "train"
```

Reply:
Indeed `path` can be one path, or a dict or a list... as it is the case for `load_dataset`...

Reply:
However, this behavior is different from the one in `load_dataset`
> split (`datasets.Split` or `str`): which split of the data to load.
> - If None, will return a `dict` with all splits (typically `datasets.Split.TRAIN` and `datasets.Split.TEST`).
> - If given, will return a single Dataset.

Reply:
It can make sense if Dataset.from_csv always returns a Dataset, not a DatasetDict with several splits.
If users want to have several splits maybe we can have DatasetDict.from_csv

Or we can allow a dictionary of paths but in this case it's better if it's consistent with load_dataset

Reply:
I see the problem... Yes, it makes sense that Dataset.from_csv should return a Dataset (and not a DatasetDict)... 

I guess we must make a decision with this trade-off between consistency and user-experience: is the user responsible for using Dataset/DatasetDict .from_csv or does the user expect (and does not mind) either one of them and the user?

Reply:
I have a small preference for having both Dataset/DatasetDict.from_csv since it's simple and clear.
Moreover the `split` argument of load_dataset is different from the one used in from_xxx:
- The split argument in load_dataset is used to select the splits you want among the available splits. Also when loading a csv with a single file as you did, only a train split is available by default.
- The split argument in Dataset.from_xxx is used to set the split name of the resulting Dataset object

Because of that I'm more in favor of implementing Dataset/DatasetDict.from_csv separately and have a different behavior than load_dataset.

Reply:
Done!